x-pack/plugins/security_solution/public/timelines/components/timeline/data_providers/add_data_provider_popover.tsx

```diff
@@ -0,0 +1,194 @@
+/*
+ * Copyright Elasticsearch B.V. and/or licensed to Elasticsearch B.V. under one
+ * or more contributor license agreements. Licensed under the Elastic License;
+ * you may not use this file except in compliance with the Elastic License.
+ */
+
+import React, { useCallback, useMemo, useState } from 'react';
+import {
+  EuiButton,
+  EuiContextMenu,
+  EuiFlexItem,
+  EuiText,
+  EuiPopover,
+  EuiIcon,
+  EuiContextMenuPanelItemDescriptor,
+} from '@elastic/eui';
+import { useDispatch, useSelector } from 'react-redux';
+
+import { TimelineType } from '../../../../../common/types/timeline';
+import { useWithSource } from '../../../../common/containers/source';
+import { StatefulEditDataProvider } from '../../edit_data_provider';
+import { addContentToTimeline } from './helpers';
+import { DataProviderType } from './data_provider';
+import { timelineSelectors } from '../../../store/timeline';
+import { ADD_FIELD_LABEL, ADD_TEMPLATE_FIELD_LABEL } from './translations';
+
+interface AddDataProviderPopoverProps {
+  timelineId: string;
+}
+
+const AddDataProviderPopoverComponent: React.FC<AddDataProviderPopoverProps> = ({ timelineId }) => {
+  const dispatch = useDispatch();
+  const [isAddFilterPopoverOpen, setIsAddFilterPopoverOpen] = useState(false);
+  const timelineById = useSelector(timelineSelectors.timelineByIdSelector);
+  const { dataProviders, timelineType } = timelineById[timelineId] ?? {};
+  const { browserFields } = useWithSource('default');
```

Comment:
be careful here, I think that you need to get the addIndex because if we select eventType === 'alerts', you will miss all the field from signals/rule

Reply:
Thank you @XavierM !